pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -258,8 +270,22 @@ func (r *MemberRollReconciler) reconcileObject(ctx context.Context, roll *maistr
 		memberStatusMap[c.Namespace] = c
 	}
 
-	// 3. create ServiceMeshMember object for each ns in spec.members that exists
-	for _, ns := range roll.Spec.Members {
+	// 3. create ServiceMeshMember object for each ns in spec.members
+	memberNamespaces := roll.Spec.Members
+	if roll.Spec.IsClusterScoped() {
+		nsList := &corev1.NamespaceList{}
+		err = r.Client.List(ctx, nsList)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		memberNamespaces = []string{}
+		for _, ns := range nsList.Items {
+			if ns.Name != meshNamespace && !isExcludedNamespace(ns.Name) {
+				memberNamespaces = append(memberNamespaces, ns.Name)
```

Comment:
won't this become very costly in large clusters? we're creating `Member` resources one at a time...

Reply:
Costly in what way?

Reply:
nevermind, my thinking was that the initial creation of these Member resource might be in big clusters. but I guess it's negligible given that we also have to create NetworkAttachmentDefinitions